aws/data_source_aws_ec2_instance_spot_price_test.go

```diff
@@ -0,0 +1,61 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccAWSEc2InstanceSpotPriceDataSource_Filter(t *testing.T) {
+	dataSourceName := "data.aws_ec2_instance_spot_price.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSEc2InstanceSpotPrice(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEc2InstanceSpotPriceDataSourceConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(dataSourceName, "instance_type"),
+				),
```

Comment:
It would be great to ensure in the testing here that the spot price and timestamp are appropriately saved

```suggestion
				Check: resource.ComposeTestCheckFunc(
					resource.TestMatchResourceAttr(dataSourceName, "spot_price", regexp.MustCompile(`^\d+\.\d+$`)),
					resource.TestMatchResourceAttr(dataSourceName, "spot_price_timestamp", regexp.MustCompile(rfc3339RegexPattern),
				),
```

Reply:
Ooh, nice, thanks for the suggestion. :+1: Added!